plugins/com.aptana.js.core/src/com/aptana/js/internal/core/node/NodePackageManager.java

```diff
@@ -489,16 +489,23 @@ public IPath getModulesPath(String packageName, boolean isGlobal, String... args
 		{
 			CollectionsUtil.addToList(processArgs, args);
 		}
-
+		
 		IStatus status = runInBackground(CollectionsUtil.toArray(processArgs));
-		if (!status.isOK())
+		IPath moduleIPath = null;
+		boolean isPathExists = false;
+		String message = status.getMessage();
+		if (!StringUtil.isEmpty(message))
+		{
+			String[] lines = message.split("\n"); //$NON-NLS-1$
+			moduleIPath = Path.fromOSString(lines[lines.length - 1]);
+			isPathExists = moduleIPath.toFile().exists();
+		}
+		if (!status.isOK() && !isPathExists) //some cases status.isNotOK even though it finds valid path
```

Comment:
To check a case where "status is notOK but having valid path" we have to go by ``!status.isOK && !isPathExists``  If we go by ``!status.isOK || !isPathExists`` this, if we take this alloy issue as an example, where ``!status.isOK`` returns true since status is not OK and it will thrown an error. If you are thinking irrespective of the state we can go just by a path existence, then we can just need to do this ``!isPathExists``